pkg/apic/apiservicerevision.go

```diff
@@ -224,9 +224,11 @@ func (c *ServiceClient) setRevisionAction(serviceBody *ServiceBody) error {
 	if serviceBody.serviceContext.serviceAction == updateAPI {
 		// Get revisions for the service and use the latest one as last reference
 		queryParams := map[string]string{
-			"query":  "metadata.references.name==" + serviceBody.serviceContext.serviceName,
-			"sort":   "metadata.audit.createTimestamp,DESC",
-			"fields": "name",
+			"query":    "metadata.references.id==" + serviceBody.serviceContext.serviceID,
+			"sort":     "metadata.audit.createTimestamp,DESC",
+			"page":     strconv.Itoa(1),
+			"pageSize": strconv.Itoa(1),
+			"fields":   "id",
 		}
 
 		revisions, err := c.GetAPIServiceRevisions(queryParams, c.cfg.GetRevisionsURL(), serviceBody.Stage)
```

Comment:
Below this the revisionCount is being calculated by counting the length of the revisions returned. If it is limited to 1 the count will never be anything else.

We may still need to return all results, not limiting the pages.

Or, we can cease to name the API Service Revisions and let API Server do the naming. Not sure if/how this will affect the Revision title, however.

Reply:
ah, good catch, let me think about that